src/components/pages/rns/sell/DomainCheckoutPage.tsx

```diff
@@ -106,169 +120,157 @@ const DomainsCheckoutPage: FC<{}> = () => {
       web3,
     },
   } = useContext(Web3Store)
-  const { dispatch: bcDispatch } = useContext(BlockchainContext)
-  const [isPendingConfirm, setIsPendingConfirm] = useState(false)
   const { dispatch: appDispatch } = useContext<AppContextProps>(AppContext)
-  const reportError = useCallback((e: UIError) => errorReporterFactory(appDispatch)(e), [appDispatch])
-
-  const currencySymbols = Object.keys(crypto)
-  const currencyOptions = currencySymbols.map((symbol) => crypto[symbol].displayName)
-
+  const reportError = useCallback(
+    (e: UIError) => errorReporterFactory(appDispatch)(e), [appDispatch],
+  )
+  const { dispatch: confirmationsDispatch } = useContext(ConfirmationsContext)
   const [price, setPrice] = useState('')
   const [priceFiat, setPriceFiat] = useState('')
   const [currency, setCurrency] = useState('0')
+  const [processingTx, setProcessingTx] = useState(false)
+  const [txOperationDone, setTxOperationDone] = useState(false)
 
-  useEffect(() => {
-    if (isPendingConfirm && order && !order.isProcessing) {
-      // Post-confirmations handle
-      const { item: { name } } = order
-      history.replace(ROUTES.RNS.SELL.DONE, { domainName: name })
-      dispatch({
-        type: 'CLEAR_ORDER',
-      } as never)
-    }
-  }, [order, isPendingConfirm, history, dispatch])
+  useEffect(() => (): void => {
+    dispatch({
+      type: 'CLEAR_ORDER',
+    })
+  }, [dispatch])
 
   if (!order) {
     history.replace(ROUTES.LANDING)
     return null
   }
 
+  const currencySymbols = Object.keys(crypto)
+  const currencyOptions = currencySymbols.map(
+    (symbol) => crypto[symbol].displayName,
+  )
+
   const {
     item: {
       name,
       expirationDate,
       tokenId,
     },
-    isProcessing,
   } = order
 
   const handleSubmit = async (): Promise<void> => {
-    if (web3 && account) {
-      dispatch({
-        type: 'SET_PROGRESS',
-        payload: {
-          isProcessing: true,
-        },
-      })
-      const currencySymbol = currencySymbols[Number(currency)] as SupportedTokenSymbol
-      const rnsContract = RNSContract.getInstance(web3, currencySymbol)
-      const marketPlaceContract = MarketplaceContract.getInstance(web3)
-      try {
-        // Get gas price
-        const gasPrice = await web3.eth.getGasPrice()
-          .catch((error: Error) => {
-            throw new UIError({
-              error,
-              id: 'web3-getGasPrice',
-              text: 'Could not retreive gas price from the blockchain.',
-            })
+    const currencySymbol = currencySymbols[Number(currency)] as SupportedTokenSymbol
+    const rnsContract = RNSContract.getInstance(web3 as Web3, currencySymbol)
+    const marketPlaceContract = MarketplaceContract.getInstance(web3 as Web3)
+    try {
+      // Get gas price
+      const gasPrice = await (web3 as Web3).eth.getGasPrice()
```

Comment:
please, change this to use the rns contract wrapper

Reply:
In a new PR may be? As this was not changed in this PR and is out of scope of confirmations handling IMO

Reply:
I disagree as the only change is a refactor:
```suggestion
      const gasPrice = await rnsContract.gasPrice()
```

That really doesn't require a separate PR. Only if you wanted to move the catch to rns, but since the function isn't used anywhere else left of the contract itself it wouldn't be a large change enough for own PR. In the master history this could appear as `refactor(rns): get price from wrapper`.